### ISSUE TITLE
[vuvuzela] Ignore scintillators (Trac #1906)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1898]
Detector simulation throws exception with 2016 GCD. This is a show stopper for the 2016 dataset. We should at least be able to ignore the scintillators for now.

ERROR (DOMLauncher): DOM OMKey(12,65,0) is of an unsupported type 40! (DOMLauncher.cxx:304 in void DOMLauncher::InitilizeDOMMap())
...

 in virtual void I3TopAddComponentWaveforms::DAQ(I3FramePtr))
FATAL (DOMLauncher): PMT pulses exist on DOM OMKey(62,65,0), but there is no entry in the DOMMap. (DOMLauncher.cxx:160 in virtual void DOMLauncher::DAQ(I3FramePtr))

Migrated from https://code.icecube.wisc.edu/ticket/1906
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:30", 
    "description": "Detector simulation throws exception with 2016 GCD. This is a show stopper for the 2016 dataset. We should at least be able to ignore the scintillators for now.\n\nERROR (DOMLauncher): DOM OMKey(12,65,0) is of an unsupported type 40! (DOMLauncher.cxx:304 in void DOMLauncher::InitilizeDOMMap())\n...\n\n in virtual void I3TopAddComponentWaveforms::DAQ(I3FramePtr))\nFATAL (DOMLauncher): PMT pulses exist on DOM OMKey(62,65,0), but there is no entry in the DOMMap. (DOMLauncher.cxx:160 in virtual void DOMLauncher::DAQ(I3FramePtr))\n\n", 
    "reporter": "juancarlos", 
    "cc": "olivas", 
    "resolution": "fixed", 
    "_ts": "1550067210114669", 
    "component": "combo simulation", 
    "summary": "[vuvuzela] Ignore scintillators", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2016-11-04T21:15:43", 
    "milestone": "", 
    "owner": "mjl5147", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20161104T21:16:32: 
* juancarlos changed attachment from "" to "2016_detector.log"
* juancarlos commented:

> Logfile

**Comment by icecube-trac** [comment]
Trac update at 20161105T12:25:28: sflis commented:

> I would argue that this is not a bug but bad input data. DOMLauncher isn't supposed to handle scintilator data, and it's probably a stretch for PMTResponse simulater as well. The solution for this is to put the scintilator MCPulses in a separate map and either do a proper PMT and electronics simulation separate from DOMLauncher or do a direct conversion to recopulses.

**Comment by icecube-trac** [comment]
Trac update at 20161105T13:31:29: 
* david.schultz changed cc from "" to "olivas"
* david.schultz commented:

> I would also ask how we got MCPulses on the scintillators. Did one of the propagators actually propagate photons as if it were a DOM? Or was it Vuvuzela adding noise?
> 
> It might actually be better for now to make the 2016 simulation GCD without scintillators or IceACT, since we don't know how to simulate them yet.

**Comment by icecube-trac** [comment]
Trac update at 20161107T20:31:11: juancarlos commented:

> I'm guessing any pulses that are there come from vuvuzela.

**Comment by icecube-trac** [comment]
Trac update at 20161114T18:21:07: kjmeagher commented:

> In changeset 151492:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="151492"
> fix wrong logic that vuvuzula used to exclude doms non-IceCube DOMs, see #1898
> ```

**Comment by icecube-trac** [comment]
Trac update at 20161114T18:27:22: kjmeagher commented:

> Vuvuzula's code which excluded non-IceCube DOMs was wrong and allowed scintillators to be simulated. But this also means that the Scintillators had non-NAN values for the noise constants in the calibration. I don't know where this new GCD files is so i can't verify it. But The GCD file should be modified so the constants are all NAN before this bug can be closed.

**Comment by icecube-trac** [comment]
Trac update at 20161206T17:00:30: 
* olivas changed status from "new" to "assigned"
* olivas changed summary from "DOMLaucher support for scintilators" to "[vuvuzela] Ignore scintillators"
* olivas changed owner from "sflis" to "mjl5147"

**Comment by icecube-trac** [comment]
Trac update at 20161215T21:43:03: 
* mjl5147 changed status from "assigned" to "accepted"
* mjl5147 changed milestone from "" to "December 2016"

**Comment by icecube-trac** [comment]
Trac update at 20170103T22:24:52: 
* mjl5147 changed resolution from "" to "fixed"
* mjl5147 commented:

> In changeset 152454:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="152454"
> Stop DOMLauncher from throwing an error when it runs into a scintillator. This didn't actually cause any issues, since only InIce and IceTop DOMs are added in the InitializeDOMMap of DOMLauncher and the modules earlier in the pipeline (including Vuvuzela) already ignore the scintillators. After Kevin's fix to Vuvuzela in r151492, this finally closes #1898.
> ```


* mjl5147 changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:30: 
* olivas changed milestone from "December 2016" to ""
* olivas commented:

> Milestone December 2016 deleted
